### PR TITLE
AAG-2740: Added Jet pack compose example app.

### PR DESCRIPTION
**Pull request by me2k**
## JIRA
[AAG-2740]

## DESCRIPTION
This PR adds a Jet Pack Compose example to the project, this uses a pre defined list just like the iOS Swift UI app and aims to be a functionally Identical app.

I have also made a couple of library updates to support JetPack Compose 1.2.0:
Gradle from 7.2 -> 7.5
Kotlin language from 1.5.19 -> 1.7.0
Gradle build tools -> 7.4.0
Kotlin Gradle plugin -> 1.7.0

## SCREENSHOTS

https://user-images.githubusercontent.com/618350/215792969-6481f0b4-bceb-4ad8-905c-58fe4a97b3fb.mov

[AAG-2740]: https://superawesomeltd.atlassian.net/browse/AAG-2740?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: superawesomecomposeexample/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        android:supportsRtl="true"
+        android:theme="@style/Theme.Samobilesdkandroid">
+        <activity
+            android:name=".activities.FeaturesActivity"
+            android:exported="true"
+            android:label="@string/app_name"
+            android:theme="@style/Theme.Samobilesdkandroid">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
```
Comment (thomas-james-1986) [inline superawesomecomposeexample/src/main/AndroidManifest.xml:22]: Perhaps an unnecessary new line here?

File: superawesomecomposeexample/src/main/java/com/superawesome/composeexample/activities/FeatureDetailActivity.kt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+class FeatureDetailActivity : ComponentActivity() {
+
+    private val featureItem: FeatureItem by lazy {
+        (intent.getParcelableExtra(featureExtraKey) as? FeatureItem)!!
+    }
+
+    private fun loadAd(placement: PlacementItem) {
+        when (featureItem.type) {
+            FeatureType.Banner -> {}
```
Comment (thomas-james-1986) [inline superawesomecomposeexample/src/main/java/com/superawesome/composeexample/activities/FeatureDetailActivity.kt:42]: Should there be a method to load a banner here?

Comment (me2k) [reply superawesomecomposeexample/src/main/java/com/superawesome/composeexample/activities/FeatureDetailActivity.kt:42]: Nope, this is deliberately left empty

File: superawesomecomposeexample/src/main/res/values/themes.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
```
Comment (thomas-james-1986) [inline superawesomecomposeexample/src/main/res/values/themes.xml:3]: Perhaps an extra new line here?

**Review by gunhansancar — APPROVED** [review]

**Review by thomas-james-1986 — COMMENTED** [review]
Looks great Myles, just a couple of minor questions!

**Review by thomas-james-1986 — APPROVED** [review]

**Pull request merged**